File: goland/internal/colours/colours.go
package colours

import "fmt"

// ANSI color codes
const (
	Red       = "\033[91m"
	Green     = "\033[92m"
	Yellow    = "\033[93m"
	Blue      = "\033[94m"
	Magenta   = "\033[95m"
	Cyan      = "\033[96m"
	White     = "\033[97m"
	Bold      = "\033[1m"
	Underline = "\033[4m"
	Reset     = "\033[0m"
)

// ColouredStr represents a string with ANSI color codes
type ColouredStr struct {
	text  string
	codes []string
}

// C creates a new ColouredStr
func C(text string) *ColouredStr {
	return &ColouredStr{
		text:  text,
		codes: []string{},
	}
}

// String returns the coloured string with ANSI codes
func (cs *ColouredStr) String() string {
	result := ""
	for _, code := range cs.codes {
		result += code
	}
	result += cs.text + Reset
	return result
}

func (cs *ColouredStr) Red() *ColouredStr       { cs.codes = append(cs.codes, Red); return cs }
func (cs *ColouredStr) Green() *ColouredStr     { cs.codes = append(cs.codes, Green); return cs }
func (cs *ColouredStr) Yellow() *ColouredStr    { cs.codes = append(cs.codes, Yellow); return cs }
func (cs *ColouredStr) Blue() *ColouredStr      { cs.codes = append(cs.codes, Blue); return cs }
func (cs *ColouredStr) Magenta() *ColouredStr   { cs.codes = append(cs.codes, Magenta); return cs }
func (cs *ColouredStr) Cyan() *ColouredStr      { cs.codes = append(cs.codes, Cyan); return cs }
func (cs *ColouredStr) White() *ColouredStr     { cs.codes = append(cs.codes, White); return cs }
func (cs *ColouredStr) Bold() *ColouredStr      { cs.codes = append(cs.codes, Bold); return cs }
func (cs *ColouredStr) Underline() *ColouredStr { cs.codes = append(cs.codes, Underline); return cs }

func (cs *ColouredStr) Print()   { fmt.Print(cs.String()) }
func (cs *ColouredStr) Println() { fmt.Println(cs.String()) }

File: python/src/oldham_quiz/colours.py
"""ANSI colour codes for terminal output."""

class Colours:
    """ANSI colour codes for terminal output."""
    RED = '\033[91m'; GREEN = '\033[92m'; YELLOW = '\033[93m'
    BLUE = '\033[94m'; MAGENTA = '\033[95m'; CYAN = '\033[96m'
    WHITE = '\033[97m'; BOLD = '\033[1m'; UNDERLINE = '\033[4m'
    RESET = '\033[0m'

    def __init__(self): self._codes = []

    @staticmethod
    def disable():
        """Disable Colours (for non-ANSI terminals)."""
        Colours.RED = ''; Colours.GREEN = ''; Colours.YELLOW = ''
        Colours.BLUE = ''; Colours.MAGENTA = ''; Colours.CYAN = ''
        Colours.WHITE = ''; Colours.BOLD = ''; Colours.UNDERLINE = ''
        Colours.RESET = ''

    def red(self): self._codes.append(self.RED); return self
    def green(self): self._codes.append(self.GREEN); return self
    def yellow(self): self._codes.append(self.YELLOW); return self
    def blue(self): self._codes.append(self.BLUE); return self
    def magenta(self): self._codes.append(self.MAGENTA); return self
    def cyan(self): self._codes.append(self.CYAN); return self
    def white(self): self._codes.append(self.WHITE); return self
    def bold(self): self._codes.append(self.BOLD); return self
    def underline(self): self._codes.append(self.UNDERLINE); return self

    def format(self, msg): return f"{''.join(self._codes)}{msg}{self.RESET}"
    def print(self, msg): print(self.format(msg))


class ColouredStr(str):
    def __getattr__(self, name):
        color_methods = {
            'red': Colours.RED, 'green': Colours.GREEN, 'yellow': Colours.YELLOW,
            'blue': Colours.BLUE, 'magenta': Colours.MAGENTA, 'cyan': Colours.CYAN,
            'white': Colours.WHITE, 'bold': Colours.BOLD, 'underline': Colours.UNDERLINE
        }
        if name in color_methods:
            return ColouredStr(f'{color_methods[name]}{self}{Colours.RESET}')
        raise AttributeError(f"'ColouredStr' object has no attribute '{name}'")

def c(text): return ColouredStr(text)

# print(c("test").red.bold)
# print(c("success").green)
